frontend/src/metabase/css/components/header.css

```diff
@@ -6,15 +6,3 @@
   font-size: 1.24em;
   color: var(--color-text-dark);
 }
-
-.Header-buttonSection {
-  padding-right: 1em;
-  margin-right: 1em;
-  border-right: 1px solid color-mod(var(--color-accent2) alpha(-80%));
-}
-
-.Header-buttonSection:last-child {
```

Comment:
Based on its usage, it would always be the last button, making these styles irrelevant